sidekick_core/lib/src/directory_ext.dart

```diff
@@ -0,0 +1,49 @@
+import 'dart:io';
+
+import 'package:dcli/dcli.dart';
+
+extension DirectoryExt on Directory {
+  /// Recursively goes up and tries to find a [Directory] matching [predicate]
+  ///
+  /// Returns `null` when reaching root (/) without a match
+  Directory? findParent(bool Function(Directory dir) predicate) {
+    var dir = this;
+    // ignore: literal_only_boolean_expressions
+    while (true) {
+      if (predicate(dir)) {
+        return dir;
+      }
+      final parent = dir.parent;
+      if (dir.toString() == parent.toString()) {
```

Comment:
This feels safer and you're using it in line 43 as well
```suggestion
      if (canonicalize(dir.path) == canonicalize(parent.path)) {
```